db/compaction/compaction_job_test.cc

```diff
@@ -390,6 +391,39 @@ class CompactionJobTestBase : public testing::Test {
                   expected_oldest_blob_file_number);
       }
     }
+
+    if (check_get_priority) {
+      CheckGetRateLimiterPriorityForWrite(compaction_job);
+    }
+  }
+
+  void CheckGetRateLimiterPriorityForWrite(CompactionJob& compaction_job) {
+    // When the state from WriteController is normal.
+    ASSERT_EQ(compaction_job.GetRateLimiterPriorityForWrite(), Env::IO_LOW);
+
+    WriteController* write_controller =
+        compaction_job.versions_->GetColumnFamilySet()->write_controller();
+
+    {
+      // When the state from WriteController is CompactionPressure.
+      std::unique_ptr<WriteControllerToken> compaction_pressure_token =
+          write_controller->GetCompactionPressureToken();
+      ASSERT_EQ(compaction_job.GetRateLimiterPriorityForWrite(), Env::IO_HIGH);
+    }
+
+    {
+      // When the state from WriteController is Delayed.
+      std::unique_ptr<WriteControllerToken> delay_token =
+          write_controller->GetDelayToken(1000000);
+      ASSERT_EQ(compaction_job.GetRateLimiterPriorityForWrite(), Env::IO_USER);
```

Comment:
Can we verify the priority flows through to the writable file?

Reply:
I can add this test in a follow up PR.